### PR TITLE
Added changes to Terraform Equinix Modules docs to include versioning, pr's, and commits

### PR DESCRIPTION
**Pull request by dcallao**
This PR contains the following changes:
- hyperlink format changes
- markdown lint throughout the doc
- added info in section 5. and 6. of the `Using the Terraform Equinix Modules Template` section

### CODE REVIEW

**Review by ctreatma — CHANGES_REQUESTED** [review]
Looks good to me overall, just spotted one broken link to the template

File: terraform-modules.md [hunk 1]
```diff
@@ -2,84 +2,131 @@
 
 Ths guide discusses how to contribute Terraform modules to the Equinix Labs GH Org.
 
-Terraform modules should either be sourced from <https://github.com/equinix-labs/terraform-equinix-template>
-or closely following the format of this project.
-
+Terraform modules should either be sourced from [Template module template] or closely following the format of this project.
```
[the remark this answers is not in the document]
Comment (dcallao) [reply terraform-modules.md:5]: The link to the template is defined right at the bottom of the page [here](https://github.com/equinix-labs/equinix-labs/pull/103/files#diff-1fc1b42aafd40dda0e7b140e3d8a0e5f9b41e024cc0dbc87bbf0aa9ad060dd3cR119). 👍

**Comment by dcallao** [comment]
@ctreatma I have updated the Terraform md doc slightly to include the CI jobs that will run once a PR is submitted and the CI jobs that will run when the PR is approved and merged.

**Review by ctreatma — APPROVED** [review]

File: terraform-modules.md [hunk 2]
```diff
@@ -2,84 +2,134 @@
...
+* [Equinix SSO]
+  * You may need to the verify networking settings
+  * You may need to Enable SSO on your [SSH key]
+* Make your membership public, if you like:
+  * [Equinix-Labs Org Members]
+* If the organization is not “equinix-labs”, “equinix”, etc, you may need to authorize the org first: [Select an Organization]
+
+2. <a name="develop"></a> Review the guidelines offered above (Steps 5-7)
 
+3. <a name="jointeam"></a> Join the [Equinis-labs Terraform team]
```
[the remark this answers is not in the document]
Comment (dcallao) [reply terraform-modules.md]: Thanks! Fixed via [7a26422](https://github.com/equinix-labs/equinix-labs/commit/7a2642225e9769cd3bec0a61e28ff14d4b1bf1de)

**Review by displague — APPROVED** [review]

**Pull request merged**